### DISCUSSION TITLE
Commit message translation

### DISCUSSION THREAD

**Discussion by popara96** [gitextensions/gitextensions#11188] [Q&A]
Hello,
Is there a way to show translated commit messages in the main window? I have a situation where a certain repo has all its commit messages written in a foreign language, and I'm wondering if there's a way to see them all automatically translated (to English) without translating them manually, one by one.
If not, what would be the scope of adding this as a feature, and would it be possible to create a nuget package for this?
Thank you!

**Comment by mstv** [comment]
GE does not have such a feature.
It would be very sensitive to performance.
Filters would even need to be translated back.
I guess it will be easier and more flexible to use a screen reader or the camera of your phone (if there is no sensitive information)...

**Comment by gerhardol** [comment]
Services such as Google Translate have API keys, but I believe all of them comes with a cost, so it does not fit well into a free application.
I assume it could be possible to let users install keys manually, but would be hard to maintain and test.

copy to browser is a better way to spend time here...

**Comment by vbjay** [comment]
Could develop a plugin that allows developers to set an api key and use a
service like google translate.

On Fri, Sep 1, 2023 at 5:32 AM popara ***@***.***> wrote:

> Closed #11188
> <https://github.com/gitextensions/gitextensions/discussions/11188> as
> resolved.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gitextensions/gitextensions/discussions/11188>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA3WDMKFSHKXIMFCVROUIDDXYGTSDANCNFSM6AAAAAA4F2QBDA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID:
> <gitextensions/gitextensions/repo-discussions/11188/discussion_event/837403
> @github.com>
>


-- 
*Jay Asbury*
PC Repair and custom programs $30/hr 1hr min
My Biking Blog <http://vbjaybiking.blogspot.com>

**Reply by popara96** [reply]
It looks to me like that code would need to be injected somewhere, in order to intercept filling the commit message text with the translated one. But yes, doesn't seem impossible, I just wonder if it'd be worth the time to develop.

**Reply by vbjay** [reply]
Would likely be a menu option that lets the user translate when desired. Plugin providing the new menu.

**Reply by gerhardol** [reply]
I believe Open link in browser is possible, then you do not have to copy manually.
